raiden-cli/src/routes/connections.ts

```diff
@@ -1,20 +1,133 @@
+  [token: string]: ApiConnection;
+}
+
+async function getConnections(this: Cli, _request: Request, response: Response) {
+  const channelsDict = await this.raiden.channels$.pipe(first()).toPromise();
+  const connections = Object.entries(channelsDict).reduce<ApiTokenConnections>(
+    (connections, [token, partnerChannel]) => ({
+      ...connections,
+      [token]: Object.values(partnerChannel).reduce<ApiConnection>(
+        ({ channels, funds }, { ownDeposit, ownWithdraw }) => {
+          const deposits = ownDeposit.sub(ownWithdraw).add(funds).toString();
+          return {
+            channels: (+channels + 1).toString(),
+            sum_deposits: deposits,
+            funds: deposits,
+          };
+        },
+        { channels: '0', sum_deposits: '0', funds: '0' },
+      ),
+    }),
+    {},
+  );
+  response.json(connections);
+}
+
+/**
+ * Naive connection manager: only supports connecting to a single hardcoded hub
+ * We don't want to improve this (for now), since SDK doesn't have the concept of connection
+ * manager; it may change when we have proper auto-pilot (#211); before that, this dummy
+ * implementation should enough to make WebUI work with it
+ */
+async function connectTokenNetwork(
+  this: Cli,
+  request: Request,
+  response: Response,
+  next: NextFunction,
+) {
+  if (request.body.initial_channel_target && request.body.initial_channel_target != 1)
+    return response.status(409).send('CLI only supports managing connection with one hub');
+  if (request.body.joinable_funds_target && request.body.joinable_funds_target != 1)
+    return response.status(409).send('CLI only supports allocating whole funds');
+  const token: string = request.params.tokenAddress;
+  try {
+    const channelsDict = await this.raiden.channels$.pipe(first()).toPromise();
+    const hub = await this.raiden.resolveName('hub.raiden.eth');
+    const channel = channelsDict[token]?.[hub];
+    if (!channel) {
+      await this.raiden.openChannel(token, hub, {
+        deposit: request.body.funds,
+      });
+    } else if (
+      request.body.funds &&
+      channel.ownDeposit.sub(channel.ownWithdraw).lt(request.body.funds)
+    ) {
+      // if channel exists but allocated funds are lower than requested, deposit difference
+      // amount = funds - (ownDeposit - ownWithdraw) = ((ownDeposit - ownWithdraw) - funds) * -1
+      const amount = channel.ownDeposit.sub(channel.ownWithdraw).sub(request.body.funds).mul(-1);
+      await this.raiden.depositChannel(token, hub, amount);
+    }
+    response.status(204);
+  } catch (error) {
+    if (isInvalidParameterError(error)) response.status(400).send(error.message);
+    else if (isInsuficientFundsError(error)) response.status(402).send(error.message);
+    else if (isConflictError(error)) response.status(409).send(error.message);
+    else next(error);
+  }
+}
+
+/**
+ * Closes all closeable channels in a token network
+ * TODO: implement auto-settle (see discussion at #237)
```

Comment:
Will this be actually a TODO of the CLI or wont it happen in the SDK exclusively?

Reply:
We could do it either implemented on the CLI or discuss #237 and possibly decide on doing it in the SDK through some `autoSettle` option. Until we decide, it remains a TODO and the users need to settle actively.